lib/mongoid/reloadable.rb

```diff
@@ -28,6 +28,9 @@ def reload
       @attributes = reloaded
       @attributes_before_type_cast = {}
       changed_attributes.clear
+      previous_changes.clear
+      previous_attributes.clear
```

Comment:
These mutate the hashes, to go with my next comment.

Reply:
Good point, thank you. Replaced `clear` calls with empty hash assignments. I think it should solve the indicated problem.